pkg/test/test.go

```diff
@@ -2,98 +2,121 @@ package test
 
 import (
 	"fmt"
-	"github.com/ory/dockertest"
+	"github.com/applike/gosoline/pkg/cfg"
 	"log"
 	"sync"
-	"time"
 )
 
-var err error
-var wait sync.WaitGroup
-var dockerPool *dockertest.Pool
-var dockerResources []*dockertest.Resource
-var configs []*ContainerConfig
-var cfgFilename = "config.test.yml"
-
-func init() {
-	dockerPool, err = dockertest.NewPool("")
-	dockerPool.MaxWait = 2 * time.Minute
-	dockerResources = make([]*dockertest.Resource, 0)
+type mockComponent interface {
+	Boot(name string, config cfg.Config, settings *mockSettings)
+	Run(runner *dockerRunner)
+	ProvideClient(clientType string) interface{}
+}
 
-	if err != nil {
-		log.Fatalf("Could not connect to docker: %s", err)
-	}
+type mockSettings struct {
+	Debug     bool   `cfg:"debug"`
+	Component string `cfg:"component"`
+	Host      string `cfg:"host"`
 }
 
-func logErr(err error, msg string) {
-	Shutdown()
-	log.Println(msg)
-	log.Fatal(err)
+type Mocks struct {
+	waitGroup    sync.WaitGroup
+	dockerRunner *dockerRunner
+	components   map[string]mockComponent
 }
 
-func Boot(configFilenames ...string) {
-	if len(configFilenames) == 0 {
-		configFilenames = append(configFilenames, cfgFilename)
-	}
+func newMocks() *Mocks {
+	components := make(map[string]mockComponent)
+	dockerRunner := newDockerRunner()
 
-	for _, filename := range configFilenames {
-		log.Println(fmt.Sprintf("booting configuration %s", filename))
-		bootFromFile(filename)
+	return &Mocks{
+		components:   components,
+		dockerRunner: dockerRunner,
 	}
+}
+
+func (t *Mocks) Boot(config cfg.Config) {
+	t.bootFromConfig(config)
 
-	wait.Wait()
+	t.waitGroup.Wait()
 
 	log.Println("test environment up and running")
-	fmt.Println()
+	log.Println()
```

Comment:
i think there is no point in keeping an empty println

Reply:
✅